drop-analytics/src/moose_impl.rs

```diff
@@ -110,17 +112,41 @@ impl MooseImpl {
         anyhow::ensure!(res.is_ok(), "Failed to initialize moose: {:?}", res.err());
 
         populate_context(&logger);
-        let res = moose::flush_changes();
-        moose_debug!(logger, res, "flush_changes");
-        res.context("Failed to flush moose context")?;
 
-        Ok(Self { logger })
+        let (tx, rx) = sync_channel(1);
+        let task_logger = logger.clone();
+        let context_update_task = std::thread::spawn(move || {
+            let mut last_update = std::time::Instant::now();
+            loop {
+                match rx.try_recv() {
+                    Ok(_) | Err(std::sync::mpsc::TryRecvError::Disconnected) => break,
+                    Err(std::sync::mpsc::TryRecvError::Empty) => {
+                        if last_update.elapsed() > Duration::from_secs(60 * 10) {
+                            populate_context(&task_logger);
+                            last_update = std::time::Instant::now();
+                        }
+                        std::thread::sleep(Duration::from_millis(100));
+                    }
+                }
+            }
```

Comment:
Oh, yeah. OS threads don't support cancelation.
But TBH should we care about the immediate cancelation?
IMHO it should look like this:
* we have an atomic bool `is_running` initialized to `true`
* we set it to `false` in the drop impl, and don't care about the joining
* in the thread we loop `while is_running`, populate the context, and then sleep for 10 min

In that way, we do not harm the battery usage with thread woken up every 100ms.
It also does not matter when we set up different moose because for sure the thread will still be alive but it won't do anything and will close itself eventually.
So one last thing is what happens when the app is closed, then the OS will simply kill the tread cleanly.

Reply:
I don't push hard for it but hope that I got you convinced

Reply:
I'm not a fan of leaving the threads dangling around like this, but you do have a very convincing point from a battery perspective that I didn't take into account, so I went with your suggestion